core/templates/pages/preferences-page/form-fields/preferred-languages.component.ts

```diff
@@ -41,8 +41,17 @@ export class PreferredLanguagesComponent {
   removable = true;
   separatorKeysCodes: number[] = [ENTER];
   formCtrl = new FormControl();
+  filteredChoices: LanguageIdAndText[] = [];
+  searchQuery: string = '';
 
-  ngOnInit(): void {
+  onInputBoxClick(): void {
+    setTimeout(() => {
+      this.searchQuery = '';
+      this.filteredChoices = this.choices;
+    });
+  }
```

Comment:
can you please add a comment on why you have used set timeout?

Reply:
SetTimeout() here ensures a smooth user experience by updating the view before modifying the searchQuery and filteredChoices.

Reply:
what error is displayed if we don't use setTimeout(). is setTimeout really required can we try to use promise or anything apart from setTimeout.

Reply:
After testing, it appears that removing setTimeout() from onInputBoxClick() doesn't cause any issues, and the view updates work fine without it

Reply:
then this is not right to add setTimeout() because of frontend tests failing, you can use the tick() & async function for solving this issue, please remove setTimeout(). `check other test files for reference`